File: src/2023-01-10/4_hamburger/solution.py
# 시간초과 실패

def solution(ingredient: list[int]) -> int:
    answer = 0

    HAMBUGER: str     = ''.join(map(str,[1, 2, 3, 1])) # <int -> str> 1231
    _ingredient: str  = ''.join(map(str, ingredient))  # 21123123...
    isLoop: bool      = True

    while(isLoop):
        findIdx: int = _ingredient.find(HAMBUGER)

        if (findIdx == -1):
            isLoop = False;
            break;
        
        answer += 1
        _ingredient = _ingredient.replace(HAMBUGER, '', 1)
        
        
    print(answer)
    return answer

  

#solution([2, 1, 1, 2, 3, 1, 1, 2, 3, 1, 2, 3, 1, 3])

File: src/2023-01-10/4_hamburger/solution.js
// https://school.programmers.co.kr/learn/courses/30/lessons/133502?language=javascript

function solution(ingredient) {
  let answer = 0;
  let stack = [];

  for (v of ingredient) {
    stack.push(v);

    if (stack.length > 3) {
      if (
        stack.at(-1) === 1 &&
        stack.at(-2) === 3 &&
        stack.at(-3) === 2 &&
        stack.at(-4) === 1
      ) {
        stack.pop();
        stack.pop();
        stack.pop();
        stack.pop();

        ++answer;
      }
    }
  }

  // console.log(answer);

  return answer;
}

solution([2, 1, 1, 2, 3, 1, 1, 2, 3, 1, 2, 3, 1, 3]);
